src/protocol/id/predicate.md

```diff
@@ -2,6 +2,5 @@
 
 For an input of type `InputType.Coin` or `InputType.Message`, `input`, the predicate owner is calculated as:
 `sha256(0x4655454C ++ CHAIN_ID ++ root(input.predicate))`, where `root` is the Merkle root of
-[the binary Merkle tree](../cryptographic_primitives.md#binary-merkle-tree) with each leaf being an 8-byte word of
-two instructions. If the bytecode is not a multiple of 8 bytes (i.e. if there are an odd number of instructions),
-the last instruction is padded with 4-byte zero.
+[the binary Merkle tree](../cryptographic_primitives.md#binary-merkle-tree) each leaf being 16KiB of instructions.
+If the bytecode is not a multiple of 16 KiB, the final leaf should be zero-padded rounding up to the nearest multiple of 8 bytes.
```

Comment:
Why multiple of 8 bytes? Do you mean padded out to a minimum of 16 KiB?

Reply:
I couldn't think of any reason why the final leaf needed to be padded out to the full leaf length, except for potentially aligning with the ethereum word size. We could save a lot of bytes on our proof if we don't need to zero-pad out to several kilobytes. So I made it a multiple of 8 bytes here to preserve the original EVM wordsize alignment.